Source/WTF/wtf/text/StringCommon.h

```diff
@@ -609,101 +607,50 @@ ALWAYS_INLINE const uint16_t* find16(const uint16_t* pointer, uint16_t character
     // If the pointer is unaligned to 16bit access, then SIMD implementation does not work. But ARM64 allows unaligned access.
     // Fallback to a simple implementation. We also use it for smaller memory where length is less than 16.
     constexpr size_t thresholdLength = 32;
-    static_assert(!(thresholdLength % (16 / sizeof(uint16_t))), "length threshold should be16-byte aligned to make find16AlignedImpl simpler");
-
-    // For first check `threshold - (unaligned >> 1)` characters, we use normal loop.
-    // This can (1) align pointer to 16-byte size so that SIMD loop gets simpler and (2) handle cases
-    // having a character in the beginning of the string efficiently.
-    uintptr_t unaligned = reinterpret_cast<uintptr_t>(pointer) & 0xf;
+    const auto* cursor = pointer;
+    const auto* end = pointer + length;
+
+    if (length >= thresholdLength) {
+        constexpr size_t stride = 16 / sizeof(UnsignedType);
+        static_assert(stride <= thresholdLength);
+        auto charactersVector = SIMD::splat(character);
+        for (; cursor + (stride - 1) < end; cursor += stride) {
+            auto value = SIMD::load(cursor);
+            auto mask = SIMD::equal(value, charactersVector);
+            if (auto index = SIMD::findFirstNonZeroIndex(mask))
+                return cursor + index.value();
+        }
```

Comment:
Previously this code first walked up to an alignment boundary -- is that hidden inside of load/equal/findFirstNonZeroIndex now?

Reply:
Previously, we are using alignment boundary, and scanning the memory regardless of the last remaining out-of-range memory (since it fits in page boundary). But this new approach is just scanning with stride and the last one is scanned with the adjusted starting point.